compiler/rustc_lint/src/context.rs

```diff
@@ -596,6 +596,9 @@ pub trait LintContext: Sized {
                     db.help("to document an item produced by a macro, \
                                   the macro must produce the documentation as part of its expansion");
                 }
+                BuiltinLintDiagnostics::PatternsInFnsWithoutBody(span) => {
+                    db.span_suggestion(span, "remove `mut` from the parameter in the trait definition", String::new(), Applicability::MachineApplicable);
```

Comment:
The suggestion here needs to be the name of the parameter, which you can pass through from `check_decl_no_pat`

Reply:
I remove `in the trait definition` message because of the conciseness